src/ucp/wireup/wireup_cm.c

```diff
@@ -1345,7 +1365,14 @@ void ucp_ep_cm_disconnect_cm_lane(ucp_ep_h ucp_ep)
     ucp_ep_update_flags(ucp_ep, UCP_EP_FLAG_DISCONNECTED_CM_LANE,
                         UCP_EP_FLAG_LOCAL_CONNECTED);
 
-    /* this will invoke @ref ucp_cm_disconnect_cb on remote side */
+    /* Remove the client's connect_progress or the server's connect notify
+     * callbacks from the callbackq to make sure it won't be invoked after
+     * CM lane has already been disconnected */
+    ucs_callbackq_remove_if(&ucp_ep->worker->uct->progress_q,
+                            ucp_cm_connect_progress_remove_filter,
+                            ucp_ep);
```

Comment:
maybe we can check LOCAL_CONNECTED flag in these callbacks instead of removing them?

Reply:
nice catch, done